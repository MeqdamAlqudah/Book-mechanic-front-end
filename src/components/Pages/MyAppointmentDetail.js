import { v4 as uuidv4 } from 'uuid';
import { useState, useEffect } from 'react';
import AxiosWrapper from '../../requirments/AxiosWrapper';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

const MyAppointmentDetail = ({userid}) => {
  const [currentAppointments, setCurrentAppointments] = useState({});
  const user = useSelector((el) => el.userReducer.current_user);
  useEffect(() => {

    AxiosWrapper(`http://127.0.0.1:3000/api/v1/users/${userid}/appointment`).then((res) => {
      setCurrentAppointments(res.data);
    });
  }, [user]);
  if (Object.keys(currentAppointments).length <= 0) {
    return (<div>loading...</div>);
  }
  return (
    <div>
      {' '}
      <h2>Appointments:</h2>
      {' '}
      <ul>
        {currentAppointments.map((element) => (

          <li key={uuidv4()}>
            Description:
            {' '}
            {element.description}
            <br />
            Date:
            {' '}
            {element.date}
            <br />
          </li>
        ))}
      </ul>
    </div>
  );
};
MyAppointmentDetail.propTypes = {
  userId: PropTypes.number,
};
MyAppointmentDetail.defaultProps = {
  userId: 0,
};
export default MyAppointmentDetail;
