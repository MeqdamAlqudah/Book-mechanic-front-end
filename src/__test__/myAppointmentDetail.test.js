import * as React from 'react';
import MyAppointmentDetail from '../components/Pages/MyAppointmentDetail';

const renderer = require('react-test-renderer');

global.React = React;
describe('Test MyAppointmentDetail', () => {
  test('test My appointment rendering ', () => {
    const v4 = jest.fn(() => {});
    const AxiosWrapper = jest.fn(() => Promise.resolve({
      json: () => Promise.resolve({
        car_id: 2,
        created_at: '2022-06-17T21:47:57.658Z',
        date: '1999-07-01T00:00:00.000Z',
        description: 'desciption for appointment',
        id: 1,
        updated_at: '2022-06-17T21:47:57.658Z',
        user_id: 1,
      }),
    }));
    const component = renderer.create(
      <MyAppointmentDetail userid={1} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
