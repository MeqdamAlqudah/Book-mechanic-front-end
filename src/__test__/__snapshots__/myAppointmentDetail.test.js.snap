// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyAppointmentDetail test My appointment rendering  1`] = `
<div>
  loading...
</div>
`;
